sechub-pds-solutions/owasp-zap/docker/scripts/owasp-zap.sh

```diff
@@ -1,11 +1,32 @@
 #!/usr/bin/env sh
 # SPDX-License-Identifier: MIT
 
-sechub_job_uuid=$(echo "$PDS_JOB_WORKSPACE_LOCATION" | cut -d "/" -f 4)
+shutdownZAP() {
+	# -f to specify the process by looking at the parameters
+	pkill -9 -f "/usr/bin/owasp-zap"
+	pkill -9 -f "/usr/share/zaproxy/zap-"
+}
 
-echo "Clean up possible old session data before starting sechub job: $sechub_job_uuid"
-rm -r ~/.ZAP/session/*
-rm ~/.ZAP/zap.log*
+# Start OWASP-ZAP server
+echo "Starting up OWASP-ZAP server"
+# -silent: disables telemetry calls, of the call home addon: https://www.zaproxy.org/docs/desktop/addons/call-home/
+#   This addon is mandatory now but the telemetry calls can be deactivated.
+#   This feature addtionally disables automated update calls, e.g. to update extensions.
+#   Otherwise, if you want to use a specific versions of extensions e.g. for testing reasons, ZAP would automatically check for updates.
+owasp-zap -daemon -silent -nostdout -host $ZAP_HOST -port $ZAP_PORT -config api.key=$ZAP_API_KEY &
+
+echo "Waiting for OWASP-ZAP to start"
+wget --quiet --output-document=- --retry-connrefused --tries=20 --waitretry=6 --header="Accept: application/json" --header="X-ZAP-API-Key: $ZAP_API_KEY" --no-check-certificate https://$ZAP_HOST:$ZAP_PORT/JSON/core/view/version
+if [ $? -ne 0 ]
+then
+    echo "OWASP-ZAP did not start after waiting for 2 minutes"
+    shutdownZAP
+    exit 1
+fi
+echo "OWASP-ZAP started"
+
+# Execute scan
+sechub_job_uuid=$(echo "$PDS_JOB_WORKSPACE_LOCATION" | cut -d "/" -f 4)
```

Comment:
The sechub_job_uuid is available as variable inside the PDS. The variable is called `SECHUB_JOB_UUID`.

See: https://mercedes-benz.github.io/sechub/latest/sechub-product-delegation-server.html#generated-variables